ftdc/custom_format.go

```diff
@@ -164,35 +175,51 @@ func flattenStruct(item reflect.Value) ([]float32, error) {
 		rField := flattenPtr(rVal.Field(memberIdx))
 		switch {
 		case rField.CanUint():
+			fields = append(fields, rType.Field(memberIdx).Name)
 			numbers = append(numbers, float32(rField.Uint()))
 		case rField.CanInt():
+			fields = append(fields, rType.Field(memberIdx).Name)
 			numbers = append(numbers, float32(rField.Int()))
 		case rField.CanFloat():
+			fields = append(fields, rType.Field(memberIdx).Name)
 			numbers = append(numbers, float32(rField.Float()))
 		case rField.Kind() == reflect.Bool:
 			if rField.Bool() {
+				fields = append(fields, rType.Field(memberIdx).Name)
 				numbers = append(numbers, 1)
 			} else {
+				fields = append(fields, rType.Field(memberIdx).Name)
 				numbers = append(numbers, 0)
 			}
 		case rField.Kind() == reflect.Struct ||
 			rField.Kind() == reflect.Pointer ||
 			rField.Kind() == reflect.Interface:
-			subNumbers, err := flattenStruct(rField)
+			subFields, subNumbers, err := flattenStructNew(rField)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
+			}
+
+			thisFieldName := rType.Field(memberIdx).Name
+			for _, subField := range subFields {
+				fields = append(fields, fmt.Sprintf("%v.%v", thisFieldName, subField))
 			}
+
 			numbers = append(numbers, subNumbers...)
 		case isNumeric(rField.Kind()):
 			//nolint:stylecheck
-			return nil, fmt.Errorf("A numeric type was forgotten to be included. Kind: %v", rField.Kind())
+			return nil, nil, fmt.Errorf("A numeric type was forgotten to be included. Kind: %v", rField.Kind())
 		default:
 			// Getting the keys for a structure will ignore these types. Such as the antagonistic
 			// `channel`, or `string`. We follow suit in ignoring these types.
 		}
 	}
 
-	return numbers, nil
+	return fields, numbers, nil
+}
+
+func getFieldsForStruct(item reflect.Value) ([]string, error) {
```

Comment:
The new `flattenStructNew` does everything `getFieldsForStruct[Old]` does and more (gets the data values). To help test correctness of the new code, I've started this PR where everything funnels into the new code.